### PR TITLE
Make ASSERTIONS=2 imply STACK_OVERFLOW_CHECK=2

### PR DESCRIPTION
**Pull request by kripken**
This is in addition to the existing 1 implying 1.

The goal is to make it easier to use `-s ASSERTIONS=2` and have it check
as many things as possible.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -7687,6 +7687,11 @@ def test_stack_overflow_check(self):
     self.emcc_args += ['-DONE_BIG_STRING']
     self.do_runf(path_from_root('tests', 'stack_overflow.cpp'), 'stack overflow', assert_returncode=NON_ZERO)
 
+    # ASSERTIONS=2 implies STACK_OVERFLOW_CHECK=2
+    self.set_setting('STACK_OVERFLOW_CHECK', 0)
```
Comment (sbc100) [inline tests/test_core.py:7691]: I think you can do `clear_setting` here instead.   That will mean its not passed at all.

**Review by sbc100 — APPROVED** [review]

File: tests/test_core.py (hunk 1 above)
Comment (kripken) [reply tests/test_core.py:7691]: Is that better or worse in this case? If the test mode sets the option, I actually do want to override that to 0, I think?

Comment (sbc100) [reply tests/test_core.py:7691]: clear_setting would remove any setting set by the test mode too.

Comment (sbc100) [reply tests/test_core.py:7691]: Actually if you explictly set `-s STACK_OVERFLOW_CHECK=0` on the command line that shouldn't that take presence over the `-s ASSERTIONS=2`?   Or if not at least it should be an error.  We don't want to ignore an explicit command line argument like that do we?

Comment (kripken) [reply tests/test_core.py]: Oh, I didn't realize that, thanks. Done.

Comment (kripken) [reply tests/test_core.py]: Oh, good point in the last comment too. Yes, it should not be ignored, fixed.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**